proxy/envoy/route.go

```diff
@@ -40,9 +41,9 @@ const (
 // buildListenerSSLContext returns an SSLContext struct.
 func buildListenerSSLContext(certsDir string) *SSLContext {
 	return &SSLContext{
-		CertChainFile:            certsDir + "/" + certChainFilename,
-		PrivateKeyFile:           certsDir + "/" + keyFilename,
-		CaCertFile:               certsDir + "/" + rootCertFilename,
+		CertChainFile:            certsDir + proxy.CertChainFilename,
```

Comment:
Would be good to be defensive - have a function to add "/" to certsDir if missing, or similar. 
In particular since before the "/" at the end of the dir was not required ( and it's usually not required in most config formats)

Reply:
Yes, but that is a good small task to do in another PR. This PR is already big and not changing the logic much.

Reply:
added `path.Join` which does the right thing